version: '3'
services:

  reverse-proxy:
    image: traefik:latest
    command:
      - --providers.docker
      - --api.insecure=true
      - --entryPoints.http.address=:80
      - --entryPoints.https.address=:443
    labels:
      - traefik.http.routers.admin.rule=Host(`traefik.docker.localhost`)
      - traefik.http.services.admin.loadbalancer.server.port=8080
    ports:
      - "80:80"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

  jupyterlab:
    image: jupyter/datascience-notebook:latest
    container_name: jupyterlab
    depends_on:
      - reverse-proxy
    labels:
      - traefik.http.routers.jupyter.rule=Host(`${JUPYTER_LAB_HOST:-jupyterlab.docker.localhost}`)
      - traefik.http.services.jupyter.loadbalancer.server.port=8888
      - traefik.http.routers.jupyter.entrypoints=http,https
    volumes:
      - ./jupyter-data:/home/jovyan/work/persist
    environment:
      JUPYTER_ENABLE_LAB: "${JUPYTER_ENABLE_LAB}"
      JUPYTER_TOKEN: "${JUPYTER_TOKEN}"
